packages/zipkin-transport-aws-sqs/src/AwsSqsLogger.js

```diff
@@ -0,0 +1,89 @@
+const AWS = require('aws-sdk');
+const {jsonEncoder: {JSON_V2}} = require('zipkin');
+const {EventEmitter} = require('events');
+
+class AwsSqsLogger extends EventEmitter {
+  constructor(options) {
+    /* eslint-disable no-console */
+    super();
+    this.log = options.log || console;
+    this.delaySeconds = options.delaySeconds || 0;
+    this.pollerSeconds = options.pollerSeconds || 100;
+    this.queue = [];
+    this.queueBytes = 0;
+    this.errorListenerSet = false;
+    this.maxPayloadSize = 0;
+    this.queueUrl = options.queueUrl;
+    if (typeof options.encoder === 'undefined' || options.encoder === JSON_V2) {
+      this.encoder = JSON_V2;
+    } else {
+      throw new Error('Unsupported encoder. Only support JSON_V2.');
+    }
```

Comment:
I think if we are only supporting V2 then we should not accept the encoder as an option. Does SQS collector support V1 @anuraaga?

Reply:
it does, but you can constrain it if you like. just make sure list encoding is used

Reply:
Agree that for new code like this, we could just stick to V2 and we won't need the dependency on the thrift encoder